packages/core/src/agent/helpers.ts

```diff
@@ -2,18 +2,23 @@ import type { Key } from '../crypto'
 import type { ConnectionRecord } from '../modules/connections'
 import type { ResolvedDidCommService } from '../modules/didcomm'
 import type { OutOfBandRecord } from '../modules/oob/repository'
+import type { BaseRecord } from '../storage/BaseRecord'
 import type { OutboundMessage, OutboundServiceMessage } from '../types'
 import type { AgentMessage } from './AgentMessage'
 
-export function createOutboundMessage<T extends AgentMessage = AgentMessage>(
-  connection: ConnectionRecord,
-  payload: T,
+export function createOutboundMessage<T extends AgentMessage = AgentMessage>(options: {
```

Comment:
It's been on my todo list for a really long time to make the outbound message a class instead of an object so we can add getters / helper methods etc.. to it. It's feel more in line with the rest of the framework, and we also use a class for the InboundMessageContext. What if we make this an `OutboundMessageContext` class instead?